api/JenkinsApi.ts

```diff
@@ -0,0 +1,43 @@
+/*
+* ---------------------------------------------------------
+* Copyright(C) Microsoft Corporation. All rights reserved.
+* ---------------------------------------------------------
+*/
+
+// Licensed under the MIT license.  See LICENSE file in the project root for full license information.
+
+/// <reference path="../node/node.d.ts"/>
+/// <reference path="../q/Q.d.ts"/>
+
```

Comment:
I'm struggling with JenkinsApi being in the agent api.  it's not part of the agent or it's api.  This seems like an internal implementation detail of your RM plugin